From 223569dc4a5778bf74f6d072b10e71912f8b5418 Mon Sep 17 00:00:00 2001
Message-ID: <223569dc4a5778bf74f6d072b10e71912f8b5418.1737052666.git.sam@gentoo.org>
In-Reply-To: <1993383ddf67e296334c7916d6afc699ee6300c7.1737052666.git.sam@gentoo.org>
References: <1993383ddf67e296334c7916d6afc699ee6300c7.1737052666.git.sam@gentoo.org>
From: Wim Taymans <wtaymans@redhat.com>
Date: Mon, 30 Sep 2024 10:06:30 +0200
Subject: [PATCH 8/8] spa: initialize all fields explicitly

Patch by Petar Popovic

Fixes #4325

(cherry picked from commit 0ca64277b317b4836beccaa3248ab9055526811c)
---
 spa/include/spa/pod/builder.h | 2 +-
 spa/include/spa/pod/parser.h  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/spa/include/spa/pod/builder.h b/spa/include/spa/pod/builder.h
index 0564d94ea..6d3e9d54a 100644
--- a/spa/include/spa/pod/builder.h
+++ b/spa/include/spa/pod/builder.h
@@ -49,7 +49,7 @@ struct spa_pod_builder {
 	struct spa_callbacks callbacks;
 };
 
-#define SPA_POD_BUILDER_INIT(buffer,size)  ((struct spa_pod_builder){ (buffer), (size), 0, {0}, {0} })
+#define SPA_POD_BUILDER_INIT(buffer,size)  ((struct spa_pod_builder){ (buffer), (size), 0, {0,0,NULL},{NULL,NULL}})
 
 static inline void
 spa_pod_builder_get_state(struct spa_pod_builder *builder, struct spa_pod_builder_state *state)
diff --git a/spa/include/spa/pod/parser.h b/spa/include/spa/pod/parser.h
index 083f91171..4c1e72acc 100644
--- a/spa/include/spa/pod/parser.h
+++ b/spa/include/spa/pod/parser.h
@@ -33,7 +33,7 @@ struct spa_pod_parser {
 	struct spa_pod_parser_state state;
 };
 
-#define SPA_POD_PARSER_INIT(buffer,size)  ((struct spa_pod_parser){ (buffer), (size), 0, {0} })
+#define SPA_POD_PARSER_INIT(buffer,size)  ((struct spa_pod_parser){ (buffer), (size), 0, {0,0,NULL}})
 
 static inline void spa_pod_parser_init(struct spa_pod_parser *parser,
 				       const void *data, uint32_t size)
-- 
2.48.0

